### PR TITLE
ci: Refactor main pipeline

### PR DESCRIPTION
**Pull request by konrad-ohms**
The main pipeline still relied on AWS code build which we are going to decommission. This PR moves the build to our Concourse Cluster and also uses skopeo to more efficiently copy images.

Signed-off-by: Konrad Ohms <konrad.ohms@de.ibm.com>

### CODE REVIEW

File: Dockerfile
Comment (Milica-Cvrkota-IBM) [inline Dockerfile:59]: Would this be equivalent to the `MAINTAINER`?

File: ci/container-image-build-task.yml
Comment (Milica-Cvrkota-IBM) [inline ci/container-image-build-task.yml:13]: Should we use the latest here? Any reason for pinning the version?

File: ci/container-image-build-task.yml
Comment (Milica-Cvrkota-IBM) [inline ci/container-image-build-task.yml:1]: Should we add copy rights to this file too?

File: ci/create-tag-file.yml
Comment (Milica-Cvrkota-IBM) [inline ci/create-tag-file.yml:3]:
```suggestion
# (c) Copyright IBM Corp. 2024
```

File: ci/pipeline.yaml
Comment (Milica-Cvrkota-IBM) [inline ci/pipeline.yaml:72]: I would vote to add a tag here as
```
tag: ((branch))
```
for all internal images. With this approach we would ensure that we can distinguish between different images in the artifactory UI. This is the approach we have in other PR pipelines

File: ci/pipeline.yaml
Comment (Milica-Cvrkota-IBM) [inline ci/pipeline.yaml:107]: should we remove this?

File: ci/pipeline.yaml
Comment (Milica-Cvrkota-IBM) [inline ci/pipeline.yaml:371]: This is just a suggestion to match the naming in the other pipelines
```suggestion
            ARTIFACTORY_USERNAME_INTERNAL: ((delivery-instana-io-internal-project-artifact-read-writer-creds.username))
            ARTIFACTORY_PASSWORD_INTERNAL: ((delivery-instana-io-internal-project-artifact-read-writer-creds.password))
```

File: ci/pipeline.yaml
Comment (Milica-Cvrkota-IBM) [inline ci/pipeline.yaml:373]: This is just a suggestion to match the naming in the other pipelines
```suggestion
            ARTIFACTORY_USERNAME_RELEASE: ((delivery-instana-io-release-project-artifact-read-writer-creds.username))
            ARTIFACTORY_PASSWORD_RELEASE: ((delivery-instana-io-release-project-artifact-read-writer-creds.password))
```

File: ci/pipeline.yaml
Comment (Milica-Cvrkota-IBM) [inline ci/pipeline.yaml:523]: Let's remove this before merging

**Review by Milica-Cvrkota-IBM — CHANGES_REQUESTED** [review]
Thanks for the changes! Besides the comments, another note we discussed in slack is to put self-update before any other job

File: Dockerfile
Comment (konrad-ohms) [reply Dockerfile:59]: yes, MAINTAINER is deprecated according to the Dockerfile reference: https://docs.docker.com/reference/dockerfile/#maintainer-deprecated

File: ci/container-image-build-task.yml
Comment (konrad-ohms) [reply ci/container-image-build-task.yml:13]: Not sure, I tend to pin versions, but latest should work just fine as well. Probably latest would not hurt here. :+1:

File: ci/container-image-build-task.yml
Comment (konrad-ohms) [reply ci/container-image-build-task.yml:1]: I'd say than we should them add to all pipeline files, but yes, I can add those.

File: ci/pipeline.yaml
Comment (konrad-ohms) [reply ci/pipeline.yaml:107]: I was actually wondering that no olm bundle image was created, but apparently, we did not need it in that pipeline so far, just the bundle yaml files. So yeah, let me remove that.

**Review by Milica-Cvrkota-IBM — APPROVED** [review]
LGTM 👍

**Pull request merged**
